Error when deploy to Google App Engine

<!-- ⚠️ If you do not respect this template your issue will be closed. -->

<!-- =============================================================================== -->
<!-- ⚠️ If you are not using the current Strapi release, you will be asked to update. -->
<!-- Please see the wiki for guides on upgrading to the latest release. -->
<!-- =============================================================================== -->

<!-- ⚠️ Make sure to browse the opened and closed issues before submitting your issue. -->

<!-- ⚠️ Before writing your issue make sure you are using:-->
<!-- Node 10.x.x -->
<!-- npm 6.x.x -->
<!-- The latest version of Strapi. -->

**Informations**
- **Node.js version**: v10.0.0
- **npm version**: v6.0.0
- **Strapi version**: v3.0.0-alpha.16
- **Database**: MongoDB
- **Operating system**: at Google Cloud


**What is the current behavior?**

After deploy the app:

```
textPayload:  "Error: Cannot find module 'immutable'
    at Function.Module._resolveFilename (module.js:548:15)
    at Function.Module._load (module.js:475:25)
    at Module.require (module.js:597:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/srv/plugins/content-type-builder/services/ContentTypeBuilder.js:7:31)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)" 
```

![screenshot 2018-10-15 at 16 51 03](https://user-images.githubusercontent.com/1007051/46958692-b4b37180-d09a-11e8-851e-d6afe10e02b2.png)


**Steps to reproduce the problem**

Deploy the app and go to https://project-id.appspot.com/

**What is the expected behavior?**

Working directly.

--- a/packages/strapi-generate-new/files/config/environments/production/server.json
+++ b/packages/strapi-generate-new/files/config/environments/production/server.json
@@ -1,6 +1,7 @@
 {
   "host": "localhost",
   "port": "${process.env.PORT || 1337}",
+  "production": true,
   "proxy": {
     "enabled": false
   },

--- a/packages/strapi-generate-new/files/config/environments/staging/server.json
+++ b/packages/strapi-generate-new/files/config/environments/staging/server.json
@@ -1,6 +1,7 @@
 {
   "host": "localhost",
   "port": "${process.env.PORT || 1337}",
+  "production": true,
   "proxy": {
     "enabled": false
   },

--- a/packages/strapi-plugin-graphql/services/Schema.js
+++ b/packages/strapi-plugin-graphql/services/Schema.js
@@ -266,7 +266,7 @@ module.exports = {
       resolvers,
     });
 
-    if (strapi.config.environment === 'development') {
+    if (!strapi.config.currentEnvironment.server.production) {
       // Write schema.
       this.writeGenerateSchema(graphql.printSchema(schema));
     }

--- a/packages/strapi-plugin-graphql/services/Schema.js
+++ b/packages/strapi-plugin-graphql/services/Schema.js
@@ -266,8 +266,10 @@ module.exports = {
       resolvers,
     });
 
-    // Write schema.
-    this.writeGenerateSchema(graphql.printSchema(schema));
+    if (strapi.config.environment === 'development') {
+      // Write schema.
+      this.writeGenerateSchema(graphql.printSchema(schema));
+    }
 
     // Remove custom scaler (like Upload);
     typeDefs = Types.removeCustomScalar(typeDefs, resolvers);

--- a/packages/strapi-plugin-users-permissions/services/UsersPermissions.js
+++ b/packages/strapi-plugin-users-permissions/services/UsersPermissions.js
@@ -438,8 +438,10 @@ module.exports = {
     try {
       // Disable auto-reload.
       strapi.reload.isWatching = false;
-      // Rewrite actions.json file.
-      fs.writeFileSync(actionsPath, JSON.stringify({ actions: data }), 'utf8');
+      if (strapi.config.environment === 'development') {
+        // Rewrite actions.json file.
+        fs.writeFileSync(actionsPath, JSON.stringify({ actions: data }), 'utf8');
+      }
       // Set value to AST to avoid restart.
       _.set(strapi.plugins['users-permissions'], 'config.actions', data);
       // Disable auto-reload.

--- a/packages/strapi-plugin-users-permissions/services/UsersPermissions.js
+++ b/packages/strapi-plugin-users-permissions/services/UsersPermissions.js
@@ -438,7 +438,7 @@ module.exports = {
     try {
       // Disable auto-reload.
       strapi.reload.isWatching = false;
-      if (strapi.config.environment === 'development') {
+      if (!strapi.config.currentEnvironment.server.production) {
         // Rewrite actions.json file.
         fs.writeFileSync(actionsPath, JSON.stringify({ actions: data }), 'utf8');
       }

--- a/packages/strapi/lib/core/plugins.js
+++ b/packages/strapi/lib/core/plugins.js
@@ -89,9 +89,9 @@ module.exports = async function() {
   }
 
   const existBuildPath = await fs.pathExists(buildPath);
-  if (strapi.config.environment !== 'development' && existBuildPath) {
+  if (strapi.config.currentEnvironment.server.production && existBuildPath) {
     return;
-  } else if (strapi.config.environment !== 'development' && !existBuildPath) {
+  } else if (strapi.config.currentEnvironment.server.production && !existBuildPath) {
     console.log('The plugins.json file is missing and the front-end cannot work without it. Please, create it first at development environment.');
   }
 

--- a/packages/strapi/lib/core/plugins.js
+++ b/packages/strapi/lib/core/plugins.js
@@ -88,8 +88,11 @@ module.exports = async function() {
     return;
   }
 
-  if (strapi.config.environment !== 'development') {
+  const existBuildPath = await fs.pathExists(buildPath);
+  if (strapi.config.environment !== 'development' && existBuildPath) {
     return;
+  } else if (strapi.config.environment !== 'development' && !existBuildPath) {
+    console.log('The plugins.json file is missing and the front-end cannot work without it. Please, create it first at development environment.');
   }
 
   // arrange system directories
@@ -97,7 +100,6 @@ module.exports = async function() {
     fs.remove(sourcePath),
 
     (async () => {
-      const existBuildPath = await fs.pathExists(buildPath);
       if (existBuildPath) {
         await fs.remove(buildPath);
       } else {

--- a/packages/strapi/lib/core/plugins.js
+++ b/packages/strapi/lib/core/plugins.js
@@ -88,6 +88,10 @@ module.exports = async function() {
     return;
   }
 
+  if (strapi.config.environment !== 'development') {
+    return;
+  }
+
   // arrange system directories
   await Promise.all([
     fs.remove(sourcePath),

